src/Altinn.App.Api/Controllers/DataController.cs

```diff
@@ -556,23 +643,27 @@ [FromRoute] Guid dataGuid
                 m.Id.Equals(dataGuid.ToString(), StringComparison.Ordinal)
             );
 
-            if (dataElement == null)
+            if (dataElement is null)
             {
                 return NotFound("Did not find data element");
             }
 
             DataType? dataType = await GetDataType(dataElement);
 
-            if (dataType == null)
+            if (dataType is null)
             {
                 string errorMsg =
                     $"Could not determine if {dataElement.DataType} requires app logic for application {org}/{app}";
                 _logger.LogError(errorMsg);
                 return BadRequest(errorMsg);
             }
-            else if (dataType.AppLogic?.ClassRef is not null)
+
+            if (
+                dataType.AppLogic?.ClassRef is not null
+                && dataType.AppLogic.DisallowUserDelete
+                && !UserHasValidOrgClaim()
+            )
```

Comment:
Is this a change in behavior, or does `DisallowUserDelete` default to `true`?

Reply:
Defaults to no value, eg `false`: https://github.com/Altinn/altinn-storage/blob/663f4b71747ab721b663d802c8506345be0eaa1f/src/Storage.Interface/Models/ApplicationLogic.cs#L48

Reply:
So in 
v8.3.5 - deleting form data is disallowed
v8.4.0 - deleting form data is allowed by default (so users who want same behavior as in v8.3.5 need to "opt in"?)

Has the change in default behavior been discussed? Not sure if this will affect anyone

Reply:
@martinothamar You could argue that `8.3.5` breaks with previous versions, and `8.4.0` restores the established behavior. 

The change has been discussed in the context of a subform, and agreed that this new implementation is desired.

Do as you will with that information 🙂